app/controllers/cart_controller.rb

```diff
@@ -0,0 +1,104 @@
+class CartController < ApplicationController
+  before_action :ensure_cart_exists
+  before_action :set_cart
+
+  def show
+    @cart_items = @cart.cart_items.includes(:product)
+  end
+
+  def create
+    product = Product.find(params[:product_id])
+    cart_item = @cart.cart_items.find_by(product: product)
+
+    if cart_item
+      cart_item.update(quantity: cart_item.quantity + 1)
+    else
+      @cart.cart_items.create(product: product, quantity: 1)
+    end
+
+    redirect_back(fallback_location: root_path, notice: "#{product.name} added to cart!")
+  end
+
+  def destroy
+    cart_item = @cart.cart_items.find(params[:id])
```

Comment:
As Florin mentioned, please note that `find` method is gonna raise an error if the record is not found.